DOCSP-35931: Issues and Help page (#2744)
jenssegers/laravel-mongodb: docs/issues-and-help.txt

.. _laravel-issues-and-help:

=============
Issues & Help
=============

We are lucky to have a vibrant PHP community that includes users of varying
experience with MongoDB PHP Library and {+odm-short+}. To get support for
general questions, search or post in the

> :community-forum:`MongoDB Community Forums <>`.

Reviewer: Q: can this link be more specific to the library? like is there a PHP tag or something

Author: yes there is - updated the link!

To learn more about MongoDB support options, see the
`Technical Support <https://www.mongodb.org/about/support>`__ page.

Bugs / Feature Requests
-----------------------

> If you've found a bug or want to see a new feature in {+odm-short+},

Reviewer: ```suggestion
If you find a bug or want to see a new feature in {+odm-short+},
```

> please report it in the GitHub issues section of the
> `mongodb/laravel-mongodb <https://github.com/mongodb/laravel-mongodb>`__
> repository.

Reviewer: ```suggestion
please report it as a GitHub issue in the
`mongodb/laravel-mongodb <https://github.com/mongodb/laravel-mongodb>`__
repository.
```

If you want to contribute code, see the following section for instructions on

> submitting pull requests.

Reviewer: S: cross link here with an anchor to the Pull Requests section

To report a bug or request a new feature, perform the following steps:

> 1. Visit the `GitHub issues <https://github.com/mongodb/laravel-mongodb/issues>`__
>    section and search for any similar issues or bugs.

Reviewer: ```suggestion
1. Review the `GitHub issues list <https://github.com/mongodb/laravel-mongodb/issues>`__
   in the source repository and search for any existing issues that address your concern.
```

.. _laravel-issues-and-help:

=============
Issues & Help
=============

We are lucky to have a vibrant PHP community that includes users of varying

> experience with MongoDB PHP Library and {+odm-short+}. To get support for

Reviewer 2: What does `{+odm-short+}` resolve to? I assume this is defined in the related repo with docs configs.

Author: Yes it's defined in [this snooty.toml file](https://github.com/10gen/docs-laravel/blob/master/snooty.toml) - it resolves to Laravel MongoDB

general questions, search or post in the
:community-forum:`MongoDB PHP Community Forums </tag/php>`.

To learn more about MongoDB support options, see the
`Technical Support <https://www.mongodb.org/about/support>`__ page.

Bugs / Feature Requests
-----------------------

If you find a bug or want to see a new feature in {+odm-short+},
please report it as a GitHub issue in the `mongodb/laravel-mongodb

> <https://github.com/mongodb/laravel-mongodb>`__ repository.

Reviewer 2: Since the repository URL is referenced a few times on this page (possibly elsewhere), would it make sense to define this somewhere so we can avoid repetition? I'm thinking similar to what exists for `:community-forum:` above, which seems to also support appending to the base URL.

Reviewer: ^ you can define a source constant for the url stem like `mongodb-laravel-gh` or something!

If you want to contribute code, see the :ref:`laravel-pull-requests` section
for instructions on submitting pull requests.

To report a bug or request a new feature, perform the following steps:

1. Review the `GitHub issues list <https://github.com/mongodb/laravel-mongodb/issues>`__
   in the source repository and search for any existing issues that address your concern.
#. If you find a matching issue, you can reply to the thread to report that
   you have a similar issue or request.
#. If you cannot find a matching issue, click :guilabel:`New issue` and select
   the appropriate issue type.

> #. If you selected "Bug report" or "Feature request", please provide as much

Reviewer: ```suggestion
#. If you select "Bug report" or "Feature request" as the issue type, please provide as much
```

   information as possible about the issue. Click :guilabel:`Submit new issue`
   to complete your submission.

If you've identified a security vulnerability in any official MongoDB
product, please report it according to the instructions found in the
:manual:`Create a Vulnerability Report page </tutorial/create-a-vulnerability-report>`.

For general questions and support requests, please use one of MongoDB's
:manual:`Technical Support </support/>` channels.

.. _laravel-pull-requests:

Pull Requests
-------------

We are happy to accept contributions to help improve {+odm-short+}.

We track current development in `PHPORM <https://jira.mongodb.org/projects/PHPORM/summary>`__
MongoDB JIRA project.

To learn more about contributing to this project, see the

> `CONTRIBUTING.md <https://github.com/mongodb/laravel-mongodb/blob/4.1/CONTRIBUTING.md>`__

Reviewer 2: I realize this repository has no named branch (e.g. main, master), so it's necessary to include "4.1" here. If you think this is something that may change across branches (e.g. the 4.2 docs would want to link to "4.2"), then perhaps this should also use a configured directive defines a different base URL (e.g. "https://github.com/mongodb/laravel-mongodb/blob/4.1/") for each branch.
